modules/javafx.controls/src/test/java/test/javafx/scene/control/SpinnerTest.java

```diff
@@ -1581,4 +1583,37 @@ public Integer fromString(String value) {
 
         assertEquals("50%", intSpinner.getEditor().getText());
     }
+
+    /**
+     * When Spinner looses focus with misformatted text in the editor,
+     * checks that the value is not changed, and the text is reverted to the value
+     */
+    @Test
+    public void testFocusLostWithTypo() {
+        Button button = new Button();
+        Spinner<Integer> spinner = new Spinner<>(new IntegerSpinnerValueFactory(0, 10, 0));
+        spinner.setEditable(true);
+
+        StageLoader stageLoader = new StageLoader(new HBox(spinner, button));
+
+        // initial value
+        spinner.getValueFactory().setValue(1);
+        int value = spinner.getValue();
+        assertEquals(1, value);
+        assertEquals("1", spinner.getEditor().getText());
+
+        // set misformatted text
+        spinner.requestFocus();
+        spinner.getEditor().setText("2abc");
+
+        // loosing focus triggers cancelEdit() because the text cannot be parsed
```

Comment:
typo: losing

Reply:
Thanks, I didn't know until today that `loose` and `lose` are actually different words in a way.🙂
I think I will still integrate it to avoid a new review cycle and given that I copied this test.
But good to know for the future!